src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp

```diff
@@ -99,6 +98,18 @@ struct typed_primitive_impl_ocl : public typed_primitive_impl<PType> {
         return make_unique<ImplType>(arg, best_kernel);
     }
 
+    template<typename ImplType>
+    static size_t get_impl_key(const typed_program_node<PType>& /*arg*/, const kernel_impl_params& impl_param) {
+        auto kernel_params = ImplType::get_kernel_params(impl_param);
+        auto params = kernel_params.first;
+#if 1
+        return params.hash();
```

Comment:
I'd expect hash to be a part of primitive, not kernel parameters. If you keep it in the kernel parameters, you'll need to implement hash function for all impl types of the same primitive, while having hash for primitive + adding impl type + shapes could be generic solution. Moreover, hash for primitive can be computed once on graph load, so in runtime you'll just need to compute hash for runtime params, and no need to convert parameters to kernel selector format (which may be expensive too)

Reply:
Thank you for your comments. I will update it.

Reply:
>> primitive + adding impl type + shape
Is it enough for the key? Even though the shape and impl type is same, there should be other parameters to be checked, like kernel parameters (padding, fused ops, etc..)